drivers/spi/spi_pw.c

```diff
@@ -211,7 +216,11 @@ static void spi_pw_rx_thld_set(const struct device *dev,
 	reg_data = (uint32_t) ~(PW_SPI_WM_MASK);
 	reg_data = PW_SPI_SIRF_WM_DFLT;
 	if (spi->ctx.rx_len && spi->ctx.rx_len < spi->fifo_depth) {
+#ifdef CONFIG_SPI_PW_LPSS_DMA
+		reg_data = 0;
+#else
 		reg_data = spi->ctx.rx_len - 1;
+#endif
```

Comment:
Seems like this could use `IS_ENABLED()`. Also, could you explain this to me: why are you completely ignoring any values that has been set to `reg_data` before this? I.e. what's the point of the three lines before the if-statement? Also (although you're not modifying them in this PR) seems like the three lines before the if-statement could simply be condensed to the third line, i.e. you could just remove the first two and nothing should change, right? (since it's overwriting whatever the two first lines did).